general:
  world_frame: "odom"
  logging_level: debug
  allow_reset_map_service: true

map:
  type: hashed_chunked_wavelet_octree
  min_cell_width: { meters: 0.01 }
  tree_height: 9

map_operations_pipeline:
  - type: threshold_map
    once_every: { seconds: 2.0 }
  - type: prune_map
    once_every: { seconds: 10.0 }
  - type: publish_map
    once_every: { seconds: 2.0 }

measurement_integrators:
  livox_mid360:
    projection_model:
      type: spherical_projector
      elevation:
        num_cells: 32
        min_angle: { degrees: -7.0 }
        max_angle: { degrees: 52.0 }
      azimuth:
        num_cells: 512
        min_angle: { degrees: -180.0 }
        max_angle: { degrees: 180.0 }
    measurement_model:
      type: continuous_beam
      angle_sigma: { degrees: 0.035 }
      # NOTE: For best results, we recommend setting range_sigma to
      #       max(min_cell_width / 2, ouster_noise) where ouster_noise = 0.05
      range_sigma: { meters: 0.05 }
      scaling_free: 0.2
      scaling_occupied: 0.4
    integration_method:
      type: hashed_chunked_wavelet_integrator
      min_range: { meters: 2.0 }
      max_range: { meters: 8.0 }
      max_update_resolution: { meters: 0.08 }

  monstar_short_range:
    projection_model:
      type: pinhole_camera_projector
      width: 352
      height: 287
      fx: 208.751708984375
      fy: 208.751708984375
      cx: 172.19403076171875
      cy: 146.1179962158203
    measurement_model:
      type: continuous_ray
      range_sigma: { meters: 0.005 }
      scaling_free: 0.4
      scaling_occupied: 0.8
    integration_method:
      type: hashed_chunked_wavelet_integrator
      min_range: { meters: 0.3 }
      max_range: { meters: 1.0 }
      max_update_resolution: { meters: 0.01 }

  monstar_medium_range:
    projection_model:
      type: pinhole_camera_projector
      width: 352
      height: 287
      fx: 208.751708984375
      fy: 208.751708984375
      cx: 172.19403076171875
      cy: 146.1179962158203
    measurement_model:
      type: continuous_ray
      range_sigma: { meters: 0.01 }
      scaling_free: 0.4
      scaling_occupied: 0.8
    integration_method:
      type: hashed_chunked_wavelet_integrator
      min_range: { meters: 1.0 }
      max_range: { meters: 2.0 }
      max_update_resolution: { meters: 0.02 }

inputs:
  - type: depth_image
    topic_name: "/pico_monstar/image_rect"
    measurement_integrator: [monstar_short_range, monstar_medium_range]
    topic_queue_length: 1
    max_wait_for_pose: { seconds: 1.0 }
    time_offset: { seconds: 0.00201747 }
    # depth_scale_factor: 0.001
    # reprojected_pointcloud_topic_name: "/wavemap/depth_cam_input"

  - type: pointcloud
    topic_name: "/livox/lidar"
    topic_type: livox
    measurement_integrator: livox_mid360
    topic_queue_length: 1
    max_wait_for_pose: { seconds: 1.0 }
    # reprojected_pointcloud_topic_name: "/wavemap/lidar_input"
    # projected_range_image_topic_name: "/wavemap/lidar_range_image"
